### PR TITLE
NodeResourceTopology: add tests for score plugin with partial node data

### PR DESCRIPTION
**Pull request by ffromani**
#### What type of PR is this?

Add test for the NodeResourceTopology score plugin to ensure correct behavior with partial data available (some nodes have NRT data, some nodes don't)

/kind cleanup

#### What this PR does / why we need it:
Since the very beginning the NRT filter plugin wanted to deal with partial data (the filter plugin silently allows nodes without NRT data!) and we need to make sure the score plugin works as expected in this scenario. Let's start with test coverage.

#### Which issue(s) this PR fixes:
Fixes N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by ffromani** [comment]
/cc @PiotrProkop 

because I'm changing the LeastNUMA testcases. Tests should behave the same, though (no intended change of behavior).

**Comment by ffromani** [comment]
/cc @Tal-or 

because he did major contributions to the score plugin

File: pkg/noderesourcetopology/score_test.go [hunk 1]
```diff
@@ -133,7 +144,7 @@ func TestNodeResourceScorePlugin(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		nodesMap, lister := initTest(defaultNUMANodes(topologyv1alpha2.SingleNUMANodeContainerLevel))
+		nodesMap, lister := initTest(defaultNUMANodes(topologyv1alpha2.SingleNUMANodeContainerLevel), nrtPassthrough)
```
Comment (Tal-or) [inline pkg/noderesourcetopology/score_test.go:147]: I believe more option would introduce in the future.
Hence, I would suggest to use functional options: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

**Review by Tal-or — COMMENTED** [review]
LGTM
one comment inside

File: pkg/noderesourcetopology/score_test.go (hunk 1 above)
Comment (Tal-or) [reply pkg/noderesourcetopology/score_test.go:147]: The comment is not in place, I was referring to `defaultNUMANodes` function

**Comment by ffromani** [comment]
/hold

to give time to all reviewers

File: pkg/noderesourcetopology/score_test.go [hunk 2]
```diff
@@ -536,6 +472,144 @@ func TestNodeResourceScorePluginLeastNUMA(t *testing.T) {
...
+			wantedRes: nodeToScoreMap{},
+			requests:  pRequests,
+			strategy:  balancedAllocationScoreStrategy,
+			nrtFilter: func(nrt *topologyv1alpha2.NodeResourceTopology) *topologyv1alpha2.NodeResourceTopology {
+				return nil
+			},
+		},
+		{
+			name:      "One node with NRT data, MostAllocated strategy",
+			wantedRes: nodeToScoreMap{"Node1": 27},
```
Comment (PiotrProkop) [inline pkg/noderesourcetopology/score_test.go:531]: it took me a while to spot you are leveraging the Go famous default value of 0 for nonexistent keys in map 😄

**Comment by ffromani** [comment]
/hold cancel

got enough reviews, thanks!

**Pull request merged**